types/yup/yup-tests.ts

```diff
@@ -594,3 +599,41 @@ yup.object<MyInterface>({
         .required(),
     arrayField: yup.array(yup.string()).required()
 });
+
+const personSchema = yup.object({
+    firstName: yup.string(), // $ExpectType StringSchema<string>
+    email: yup
+        .string()
+        .nullable()
+        .email(),
+    birthDate: yup
+        .date()
+        .nullable()
+        .min(new Date(1900, 0, 1)),
+    canBeNull: yup.string().nullable(true), // $ExpectType StringSchema<string | null>
+    mustBeAString: yup
+        .string()
+        .nullable(true)
+        .nullable(false),
+    children: yup.array(yup.string()).nullable()
+});
+
+type Person = ReturnType<typeof personSchema.cast>;
+// Equivalent to:
+// type Person = {
+//     firstName: string;
+//     email: string | null;
+//     birthDate: Date | null;
+//     canBeNull: string | null;
+//     mustBeAString: string;
+//     children: string[] | null;
+// }
+
+const person: Person = {
+    firstName: "",
+    email: null,
+    birthDate: null,
+    canBeNull: null,
+    mustBeAString: "",
+    children: null
+};
```

Comment:
Would be nice to also ensure the properties can still be non-null, for example with some assignments like:
```
person.email = 'some@email.com';
person.birthDate = new Date();
person.children = ['1', '2', '3'];
```

Reply:
👍